### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by amejid**
## Milestone 2: Processing data in models

In this PR, the following were implemented:-

- Created relationships between models
- Implemented custom methods

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
+
+  def return_recent_comments
+    comment.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```
Comment (gemmen29) [inline app/models/post.rb:16]: - [ ] [OPTIONAL] Kindly use `after_save` to call `update_posts_counter` and make the `update_posts_counter` method private. The `after_save` will call this method each time a new post is saved in the database.

File: app/models/like.rb [hunk 2]
```diff
@@ -1,2 +1,10 @@
 class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  private
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (gemmen29) [inline app/models/like.rb:11]: - [ ] [OPTIONAL] Please do the same here

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,2 +1,10 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  private
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (gemmen29) [inline app/models/comment.rb:11]: - [ ] [OPTIONAL] Please do the same here

**Review by gemmen29 — APPROVED** [review]
Hi @amejid, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions:
- [ ]  _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (amejid) [reply app/models/post.rb:16]: Noted.

File: app/models/like.rb (hunk 2 above)
Comment (amejid) [reply app/models/like.rb:11]: Noted.

File: app/models/comment.rb (hunk 3 above)
Comment (amejid) [reply app/models/comment.rb:11]: Noted.

**Comment by amejid** [comment]
Thanks for reviewing my code and giving me constructive feedback.

**Pull request merged**